Authentication/controllers/authControllers.js

```diff
@@ -0,0 +1,132 @@
+const { pool } = require('../models/configDB')
+const bcrypt = require("bcrypt");
+const jwt = require('jsonwebtoken');
+const JWT_SECRET = process.env.JWT_SECRET
+
+const login = async (req, res) => {
+    let success = false;
+
+    const { email, password } = req.body;
+    pool.query(
+        `SELECT * FROM users WHERE email = $1`,
+        [email],
+        (err, results) => {
+            if (err) {
+                throw err;
+            }
+
+            if (results.rows.length > 0) {
+
+                const user = results.rows[0];
+                bcrypt.compare(password, user.user_password, (err, isMatch) => {
+                    if (err) {
+                        console.log("error:", err);
+                    }
+                    if (isMatch) {
+                        const data = {
+                            user: {
+                                id: user.user_id
+                            }
+                        }
+                        const authToken = jwt.sign(data, JWT_SECRET);
+                        success = true
+                        res.json({ success, authToken })
+                    } else {
+                        //password is incorrect
+                        success = false
+                        return res.status(400).json({ success, error: "Please try to login with correct credentials" });
+                    }
+                });
+            } else {
+                // No user
+                return done(null, false, {
+                    message: "No user with that email address"
+                });
+            }
+        }
+    );
+}
+
+const getUserData = async (req, res) => {
```

Comment:
why is getUserData part of authentication? 
I think you can remove it

Reply:
then how we will get User Data and like say display his profile, name or other things when the user logs in

Reply:
That's what I'm saying, it shouldn't be in auth